pkg/integrity/dsse.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the LICENSE.md file
+// distributed with the sources of this project regarding your rights to use or distribute this
+// software.
+
+package integrity
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+)
+
+var metadataMediaType = "application/vnd.sylabs.sif-metadata+json"
```

Comment:
Not really related to this, but it jogged my brain a bit... I think it'd be a good at some point soon to add an .md file somewhere in the SIF repo that lists mediatypes in use (in Singularity ORAS code as well as here).

Might be good to bring the mediatypes out of Singularity into SIF as constants?

Reply:
Makes sense to me. Opened #244 to capture that.